v3/src/models/formula/functions/lookup-functions.ts

```diff
@@ -3,78 +3,103 @@ import { FormulaMathJsScope } from "../formula-mathjs-scope"
 import { DisplayNameMap, ILookupDependency, isConstantStringNode, rmCanonicalPrefix } from "../formula-types"
 import { UNDEF_RESULT, equal, evaluateNode } from "./function-utils"
 import type { IDataSet } from "../../data/data-set"
+import t from "../../../utilities/translation/translate"
+
+type LookupStringContantArg = ConstantNode<string> | undefined
 
 export const lookupFunctions = {
   // lookupByIndex("dataSetName", "attributeName", index)
   lookupByIndex: {
-    validateArguments: (args: MathNode[]): [ConstantNode<string>, ConstantNode<string>, MathNode] => {
-      if (args.length !== 3) {
-        throw new Error(`lookupByIndex function expects exactly 3 arguments, but it received ${args.length}`)
-      }
-      if (!isConstantStringNode(args[0]) || !isConstantStringNode(args[1])) {
-        throw new Error("lookupByIndex function expects first two arguments to be strings " +
-          "and the third one to be numeric (or evaluate to number)")
-      }
-      return [args[0], args[1], args[2]]
-    },
     getDependency: (args: MathNode[]): ILookupDependency => {
-      const validArgs = lookupFunctions.lookupByIndex.validateArguments(args)
       return {
         type: "lookup",
-        dataSetId: rmCanonicalPrefix(validArgs[0].value),
-        attrId: rmCanonicalPrefix(validArgs[1].value),
+        dataSetId: rmCanonicalPrefix((args[0] as LookupStringContantArg)?.value),
+        attrId: rmCanonicalPrefix((args[1] as LookupStringContantArg)?.value),
       }
     },
     canonicalize: (args: MathNode[], displayNameMap: DisplayNameMap) => {
-      const validArgs = lookupFunctions.lookupByIndex.validateArguments(args)
-      const dataSetName = validArgs[0].value
-      const attrName = validArgs[1].value
-      validArgs[0].value = displayNameMap.dataSet[dataSetName]?.id
-      validArgs[1].value = displayNameMap.dataSet[dataSetName]?.attribute[attrName]
+      const dataSetNameArg = args[0] as LookupStringContantArg
+      const attrNameArg = args[1] as LookupStringContantArg
```

Comment:
Could also use:
```suggestion
      const [dataSetNameArg, attrNameArg] = args as LookupStringConstantArg[]
```
And same could be done in other similar functions. Note type fix as well.

Reply:
Thanks, fixed in the last commit.